lib/asciidoctor-pdf/converter.rb

```diff
@@ -188,7 +188,7 @@ def convert_document doc
     end if respond_to? :on_page_create
 
     layout_cover_page :front, doc
-    layout_title_page doc
+    layout_title_page doc if doc.doctype == 'book' || (doc.attr? 'title-page')
```

Comment:
I would add this condition to layout_title_page function itself.
There is even FIXME comment there.

  # FIXME only create title page if doctype=book!
  def layout_title_page doc
    # Do not create title page 
    # 1. if doctype!=book
    # 2. there is no :title-page: attribute
    # 3. there is :notitle: attribute
    # 4. there is no header in the document (= header)
    if doc.doctype != 'book' && !(doc.attr? 'title-page') || doc.notitle || !doc.header?
      return
    end
...

Reply:
I'd rather keep it as it is. Whether layout_title_page is called depends on the doctype or the title-page attribute override (which can easily be set in an extension). The rest of the logic should be handled by the method itself.

There was just a time when we didn't have control over it, so the comment is actually outdated now.